pkg/runner/runner.go

```diff
@@ -309,23 +312,27 @@ func (r *Runner) deployAssets() error {
 		if err != nil {
 			return err
 		}
-		listenerHostName = gatewayv1beta1.Hostname("*.gwapi." + ingressDomain)
-		httproutes.Spec.Hostnames = append(httproutes.Spec.Hostnames, gatewayv1beta1.Hostname("nginx.gwapi."+ingressDomain))
-		log.Debugf("Creating GatewayClass...")
-		_, err = hrClientSet.GatewayV1beta1().GatewayClasses().Create(context.TODO(), gatewayClass, metav1.CreateOptions{})
-		if err != nil && !errors.IsAlreadyExists(err) {
+		listenerHostName = gwv1.Hostname("*.gwapi." + ingressDomain)
+		httproutes.Spec.Hostnames = append(httproutes.Spec.Hostnames, gwv1.Hostname("nginx.gwapi."+ingressDomain))
+		svc, err := clientSet.CoreV1().Services(r.igNamespace).Get(context.TODO(), r.gwLb, metav1.GetOptions{})
+		if err != nil {
 			return err
 		}
-		time.Sleep(5 * time.Second) // wait for ServiceMeshControlPlane to be ready
-		log.Debugf("Creating Gateway...")
-		_, err = hrClientSet.GatewayV1beta1().Gateways(string(gatewayNamespace)).Create(context.TODO(), &gateway, metav1.CreateOptions{})
+		gateway.Spec.Addresses = append(gateway.Spec.Addresses, gwv1.GatewayAddress{
+			Type:  ptr.To(gwv1.HostnameAddressType),
+			Value: svc.Name,
+		})
+		gateway.Spec.GatewayClassName = gwv1.ObjectName(r.gwClassController)
+		log.Debugf("Creating Gateway bound to gateway class %s", r.gwClassController)
+		_, err = hrClientSet.GatewayV1().Gateways(r.igNamespace).Create(context.TODO(), &gateway, metav1.CreateOptions{})
 		if err != nil && !errors.IsAlreadyExists(err) {
 			return err
 		}
-		log.Debugf("Waiting 4 minutes for Gateway to be ready...")
-		time.Sleep(4 * time.Minute)
+		if err := waitForGateway(r.igNamespace, gateway.Name, 4*time.Minute); err != nil {
+			return err
+		}
 		log.Debugf("Creating HTTPRoute...")
-		_, err := hrClientSet.GatewayV1beta1().HTTPRoutes(routesNamespace).Create(context.TODO(), &httproutes, metav1.CreateOptions{})
+		_, err = hrClientSet.GatewayV1().HTTPRoutes(r.igNamespace).Create(context.TODO(), &httproutes, metav1.CreateOptions{})
```

Comment:
s/r.igNamespace/routesNamespace
to create the httproutes in ingress-perf instead of openshift-ingress